pydata_center/cli/controller_cli.py

```diff
@@ -0,0 +1,280 @@
+
+            print('Waiting for result...')
+            time.sleep(interval)
+
+        except requests.RequestException as e:
+            print(f'Request failed: {e}')
+            return None
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description=(
+            'CLI for interaction with Django Controller '
+            'to trigger commands on agents.'
+        )
+    )
+    subparsers = parser.add_subparsers(dest='command')
+
+    subparsers.add_parser('agents', help='List active agents')
+
+    login_parser = subparsers.add_parser(
+        'login', help='Login and store credentials'
+    )
+    login_parser.add_argument('--username', required=True, help='Username')
+    login_parser.add_argument('--password', required=True, help='Password')
+
+    send_parser = subparsers.add_parser('send', help='Send command to agent')
+    send_parser.add_argument('hostname', help="Agent's hostname")
+    send_parser.add_argument('cmd', help='Command to send to agent')
+    send_parser.add_argument(
+        '--poll', help='Poll for result after sending', action='store_true'
+    )
+
+    poll_parser = subparsers.add_parser('poll', help='Poll result of command')
+    poll_parser.add_argument('id', help='Command ID', type=int)
+
+    args = parser.parse_args()
+
+    if args.command == 'login':
+        login(args.username, args.password)
+
+    elif args.command == 'agents':
+        auth = get_auth_from_env()
+        if not auth:
+            print("You must login first using the 'login' command.")
+            return
+        list_agents(username=auth[0], password=auth[1])
+
+    elif args.command == 'send':
+        auth = get_auth_from_env()
+        if not auth:
+            print("You must login first using the 'login' command.")
+            return
+        command_id = send_command(
+            args.hostname,
+            args.cmd,
+            username=auth[0],
+            password=auth[1]
+        )
+        if command_id:
+            print(f'Use this ID to poll the result: {command_id}')
+            if args.poll:
+                result = poll_result(
+                    command_id,
+                    interval=default_interval,
+                    timeout=default_timeout,
+                    username=auth[0],
+                    password=auth[1]
+                )
+                if result:
+                    print(
+                        'Result:',
+                        result.get('result', '[no result returned]')
+                    )
+
+    elif args.command == 'poll':
+        auth = get_auth_from_env()
+        if not auth:
+            print("You must login first using the 'login' command.")
+            return
+        result = poll_result(
+            args.id,
+            username=auth[0],
+            password=auth[1]
+        )
+        if result:
+            print('Result:', result.get('result', '[no result returned]'))
+
+    else:
+        parser.print_help()
```

Comment:
Logic is embedded in `main()` and hard to unit test. You may extract command handlers (`handle_login`, `handle_send`, etc.) to allow independent testing.